device/acpi_pm_timer.go

```diff
@@ -0,0 +1,53 @@
+package device
+
+import (
+	"encoding/binary"
+	"time"
+)
+
+type ACPIPMTimer struct {
+	Port  uint64
+	Start time.Time
+}
+
+const (
+	pmTimerFreqHz  uint64 = 3_579_545
+	nanosPerSecond uint64 = 1_000_000_000
+)
+
+func NewACPIPMTimer() *ACPIPMTimer {
+	return &ACPIPMTimer{
+		Port:  0x608,
+		Start: time.Now(),
+	}
+}
+
+func (a *ACPIPMTimer) Read(base uint64, data []byte) error {
+	if len(data) != 4 {
+		return errDataLenInvalid
+	}
+
+	since := time.Since(a.Start)
+	nanos := since.Nanoseconds()
+	counter := (nanos * int64(pmTimerFreqHz)) / int64(nanosPerSecond)
```

Comment:
The timer value calculation seems to be correct.
But, I cannot comment on its accuracy.
Maybe this is enough for the current implementation.

Reply:
This device is only required for ACPI in EDK2 anyway. And the test will show, that it boots. Good enough I guess. :)